code/game/machinery/teleporter.dm

```diff
@@ -290,13 +290,27 @@
 		return
 	return ..()
 
+/obj/machinery/teleport/hub/proc/blockExiles(atom/movable/AM)
+	var/obj/machinery/computer/teleporter/com = power_station.teleporter_console
+	if(z == ZLEVEL_STATION) //if we're on the station
+		return FALSE //don't block teleportation
+	if(com.target.z == z) //if we're teleporting to the same zlevel
+		return FALSE //don't block teleportation
+	if(istype(AM, /mob/living/carbon))
+		var/mob/living/carbon/C = AM
+		for(var/obj/item/weapon/implant/exile/E in C.implants)//Checking that there is an exile implant
+			AM << "\black [src] has detected your implant and is blocking your entry."
```

Comment:
The duplication of this check is also bad, don't duplicate this create a generic proc in helpers.dm